src/compiling/lexing/lexer.rs

```diff
@@ -28,6 +32,7 @@ pub struct Lexer {
     pub scoping_mode: ScopingMode,
 }
 
+#[cfg_attr(feature = "serde", derive(Serialize))]
```

Comment:
Should this be `derive(Serialize, Deserialize)` like all the other `cfg_attr` annotations?